Database projects - add libvirt support

PaulNeumann opened the pull request:
Adds support for libvirt/KVM to the single-instance database projects.

- `Vagrantfile`s: add libvirt configuration (except for 19.3.0, which already had it)
- `README.md` files: updates for new functionality
- `OracleDatabase/README.md`: updates for new functionality and repo name; resolve markdownlint warnings

These changes have been tested on both VirtualBox and libvirt/KVM.

I'm happy to make any changes that you'd like.

Signed-off-by: Paul Neumann <38871902+PaulNeumann@users.noreply.github.com>

AmedeeBulle commented:
@PaulNeumann thank you for the contribution.

The `libvirt`changes look good to me.

Can you also replace the term `box` by `VM` where we talk about the running VM, not the base box (like in `You can shut down the box via the usual...`)?

PaulNeumann commented:
@AmedeeBulle Will do. It may be tomorrow morning (in the US) before I can work on it.

PaulNeumann commented:
@AmedeeBulle I've made the following changes to the `README.md` files:

- Replaced the term "box" with "VM" when referring to the instantiated VM
- Replaced the word "folder" with "directory" where appropriate to be consistent with other projects in this repository
- Corrected formatting in the description of the VM_MEMORY configuration parameter

I'm happy to make any additional changes that you'd like.
